libsolidity/formal/SMTEncoder.cpp

```diff
@@ -3162,11 +3162,11 @@ void SMTEncoder::collectFreeFunctions(set<SourceUnit const*, ASTNode::CompareByI
 				auto contract = dynamic_cast<ContractDefinition const*>(node.get());
 				contract && contract->isLibrary()
 			)
-			{
 				for (auto function: contract->definedFunctions())
-					if (!function->isPublic())
+					// We need to add public library functions too because they can be called
+					// internally by internal library functions that are considered free functions.
+					if (!function->isPublic() || function->libraryFunction())
```

Comment:
I think the condition can now be simplified to `true`.

Reply:
why?

Reply:
Because all these functions are library functions due to the check in line 3163

Reply:
true, changed